### PR TITLE
added the argument keepOrigNames to the function extractVariableGroup…

### PR DESCRIPTION
**Pull request by giannou**
…s to allow usage of the function in summation checks. exported the function. wrote tests

### CODE REVIEW

File: R/extractVariableGroups.R [hunk 1]
```diff
@@ -3,20 +3,27 @@
...
+extractVariableGroups <- function(x,keepOrigNames=FALSE) {
+  tmp <- function(x,sep="|+|",ext="",allVars) {
     y <- strsplit(x,sep, fixed=TRUE)
     out <- list()
     for(j in 1:length(y)) {
       for(i in 1:(length(y[[j]])-1)) {
         name <- paste0(paste(y[[j]][1:i],collapse=sep),ext)
+        ind<-grep(gsub("\\|","\\\\|",paste0("^",name,"$")),
+                  sub("\\|\\+\\+|\\|\\+","",sub(" \\(.*.\\)$","",allVars)))
+        if (keepOrigNames ) try(name<-allVars[[ind]],silent = T)
```
Comment (tscheypidi) [inline R/extractVariableGroups.R:26]: keepOrigNames is missing in the argument list of tmp

File: R/extractVariableGroups.R [hunk 2]
```diff
@@ -28,7 +35,8 @@ extractVariableGroups <- function(x) {
     matches <- grep(sep,x,fixed=TRUE, value = TRUE)
     if(length(matches)==0) break()
     ext <- ifelse(i>1,paste0(" ",i),"")
-    out <- c(out,tmp(matches,sep=sep,ext=ext))
+    if (keepOrigNames) ext<-""
```
Comment (tscheypidi) [inline R/extractVariableGroups.R:38]: perhaps move that into the tmp function to have all modifications related to keepOrigNames in one place

File: .buildlibrary [hunk 3]
```diff
@@ -1,6 +1,7 @@
-ValidationKey: '23354100'
+ValidationKey: '23718520'
 AcceptedWarnings:
 - 'Warning: package ''.*'' was built under R version'
 - 'Warning: namespace ''.*'' is not available and has been replaced'
 - ‘qpdf’ is needed for checks on size reduction of PDFs
 AcceptedNotes: unable to verify current time
+AutocreateReadme: yes
```
Comment (tscheypidi) [inline .buildlibrary:7]: If possible, can you please also add github actions?

**Review by tscheypidi — CHANGES_REQUESTED** [review]

File: R/extractVariableGroups.R (hunk 1 above)
Comment (giannou) [reply R/extractVariableGroups.R:27]: I added github actions and get this error: https://github.com/giannou/madrat/runs/1934783668?check_suite_focus=true
Any idea what the problem might be?

File: .buildlibrary (hunk 3 above)
Comment (giannou) [reply .buildlibrary:7]: looks like it's worked: https://github.com/giannou/mip/actions/runs/581231478

**Comment by giannou** [comment]
pushed a new version

**Review by tscheypidi — APPROVED** [review]

File: tests/testthat/test-extractVariableGroups.R [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+              "a 3"=c("a|+++|ble" ,"a|+++|blet"))
+
+  res2 <- list("a"=c("a|+|1|+|aa","a|+|2|abc","a|+|1|+|bb","a|+|1|+|cc","a|+|3|+|aa","a|+|3|+|bb","a|+|3"),
+               "a|+|1"=c("a|+|1|+|aa","a|+|1|+|bb","a|+|1|+|cc"),
+               "a|+|3"=c("a|+|3|+|aa","a|+|3|+|bb"),
+               "a2"=c("a2|+|bla","a2|+|blu"),
+               "a3"=c("a3|++|ble" ,"a3|++|blet"))
+  
+  expect_mapequal(extractVariableGroups(x1),res1)
+  expect_mapequal(extractVariableGroups(x2,keepOrigNames = T),res2)
```
Comment (tscheypidi) [inline tests/testthat/test-extractVariableGroups.R:44]: why do you perform the second test with x2 instead of x1? should work with x1 as well, right?

Comment (giannou) [reply tests/testthat/test-extractVariableGroups.R:44]: no, `res1` has `a 2` and `a 3` as group names, these are the automatically generated names for groups whose total is missing

Comment (tscheypidi) [reply tests/testthat/test-extractVariableGroups.R:44]: here is the result that should come out for `keepOrigNames = TRUE` for ? `x1`:

```
> extractVariableGroups(x1, keepOrigNames = TRUE)
$a
[1] "a|+|1|+|aa" "a|+|2|abc"  "a|+|1|+|bb" "a|+|1|+|cc" "a|+|3|+|aa" "a|+|3|+|bb" "a|+|3"     

$`a|+|1`
[1] "a|+|1|+|aa" "a|+|1|+|bb" "a|+|1|+|cc"

$`a|+|3`
[1] "a|+|3|+|aa" "a|+|3|+|bb"

$a
[1] "a|++|bla" "a|++|blu"

$a
[1] "a|+++|ble"  "a|+++|blet"
```
Is that still the case?

**Pull request merged**